v2/protegrity-data-tokenization/src/main/java/com/google/cloud/teleport/v2/transforms/io/GcsIO.java

```diff
@@ -173,123 +173,156 @@ public GcsIO(ProtegrityDataTokenizationOptions options) {
+                  .to(options.getOutputGcsDirectory()).withHeader(header));
+    }
+  }
+
+  private POutput writeAvro(PCollection<Row> outputCollection, Schema schema) {
+    org.apache.avro.Schema avroSchema = AvroUtils.toAvroSchema(schema);
+    return outputCollection
+        .apply(
+            "RowToGenericRecord", MapElements.into(TypeDescriptor.of(GenericRecord.class))
+                .via(AvroUtils.getRowToGenericRecordFunction(avroSchema)))
+        .setCoder(AvroCoder.of(GenericRecord.class, avroSchema))
+        .apply("WriteToAvro", AvroIO.writeGenericRecords(avroSchema)
+            .to(options.getOutputGcsDirectory())
+            .withSuffix(".avro"));
+  }
+
+  public PCollection<Row> read(Pipeline pipeline, SchemasUtils schema) {
+    switch (options.getInputGcsFileFormat()) {
+      case CSV:
+        /*
+         * Step 1: Read CSV file(s) from Cloud Storage using {@link CsvConverters.ReadCsv}.
+         */
+        PCollectionTuple csvLines = readCsv(pipeline);
+        /*
+         * Step 2: Convert lines to Json.
+         */
+        PCollectionTuple jsons = csvLineToJson(csvLines, schema.getJsonBeamSchema());
+
+        if (options.getNonTokenizedDeadLetterGcsPath() != null) {
+          /*
+           * Step 3: Write jsons to dead-letter gcs that were successfully processed.
+           */
+          jsons.get(PROCESSING_DEADLETTER_OUT)
+              .apply("WriteCsvConversionErrorsToGcs",
+                  ErrorConverters.WriteErrorsToTextIO.<String, String>newBuilder()
+                      .setErrorWritePath(options.getNonTokenizedDeadLetterGcsPath())
+                      .setTranslateFunction(SerializableFunctions.getCsvErrorConverter())
+                      .build());
+        }
+        /*
+         * Step 4: Get jsons that were successfully processed.
+         */
+        return jsons.get(PROCESSING_OUT)
+            .apply(
+                "GetJson",
+                MapElements.into(TypeDescriptors.strings()).via(FailsafeElement::getPayload))
+            .apply(new JsonToBeamRow(options.getNonTokenizedDeadLetterGcsPath(), schema));
+      case JSON:
+        return readJson(pipeline)
+            .apply(new JsonToBeamRow(options.getNonTokenizedDeadLetterGcsPath(), schema));
+      case AVRO:
+        return readAvro(pipeline, schema.getBeamSchema());
+      default:
+        throw new IllegalStateException(
+            "No valid format for input data is provided. Please, choose JSON or CSV or AVRO.");
+
+    }
+  }
+
+  public POutput write(PCollection<Row> input, Schema schema) {
+    switch (options.getOutputGcsFileFormat()) {
+      case JSON:
+        return writeJson(input);
+      case AVRO:
+        return writeAvro(input, schema);
+      case CSV:
+        return writeCsv(input, schema.getFieldNames());
+      default:
+        throw new IllegalStateException(
+            "No valid format for output data is provided. Please, choose JSON or CSV or AVRO.");
```

Comment:
Please, change the message to this:
"No valid format for output data is provided. Please, choose JSON, CSV, or AVRO."